packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -1607,44 +1607,83 @@ void main() {
-              controller: controller,
-            ),
+    final TextEditingController controller = TextEditingController(
+      text: 'Atwater Peel Sherbrooke Bonaventure',
+    );
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoTextField(
+            controller: controller,
           ),
         ),
-      );
+      ),
+    );
 
-      final Offset textfieldStart = tester.getTopLeft(find.byType(CupertinoTextField));
+    final Offset textfieldStart = tester.getTopLeft(find.byType(CupertinoTextField));
+
+    const int pointerValue = 1;
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.downWithCustomEvent(
+      textfieldStart + const Offset(150.0, 5.0),
+      PointerDownEvent(
+        pointer: pointerValue,
+        position: textfieldStart + const Offset(150.0, 5.0),
+        pressure: 3.0,
+        pressureMax: 6.0,
+        pressureMin: 0.0
+      ),
+    );
+    // We expect the force press to select a word at the given location.
+    expect(
+      controller.selection,
+      const TextSelection(baseOffset: 8, extentOffset: 12),
+    );
+
+    await gesture.up();
+    await tester.pump(const Duration(milliseconds: 500));
+    expect(find.byType(CupertinoButton), findsNWidgets(3));
+  });
 
-      const int pointerValue = 1;
-      final TestGesture gesture = await tester.createGesture();
-      await gesture.downWithCustomEvent(
-        textfieldStart + const Offset(150.0, 5.0),
-        PointerDownEvent(
-          pointer: pointerValue,
-          position: textfieldStart + const Offset(150.0, 5.0),
-          pressure: 3.0,
-          pressureMax: 6.0,
-          pressureMin: 0.0
+  testWidgets('force press on unsupported devices falls back to tap', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController(
+      text: 'Atwater Peel Sherbrooke Bonaventure',
+    );
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoTextField(
+            controller: controller,
+          ),
         ),
-      );
-      // We expect the force press to select a word at the given location.
-      expect(
-        controller.selection,
-        const TextSelection(baseOffset: 8, extentOffset: 12),
-      );
+      ),
+    );
 
-      await gesture.up();
-      await tester.pumpAndSettle();
-      expect(find.byType(CupertinoButton), findsNWidgets(3));
-    },
-  );
+    final Offset textfieldStart = tester.getTopLeft(find.byType(CupertinoTextField));
+
+    const int pointerValue = 1;
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.downWithCustomEvent(
+      textfieldStart + const Offset(150.0, 5.0),
+      PointerDownEvent(
+        pointer: pointerValue,
+        position: textfieldStart + const Offset(150.0, 5.0),
+        // iPhone 6 and below report 0 across the board.
+        pressure: 0,
+        pressureMax: 0,
```

Comment:
Wouldn't you also want to test values like 0.5 and -1.0?

Reply:
Good point. Though not in this class. Added to force_press_test.dart.